crates/plugins/mediator-coordination/src/forward/routing.rs

```diff
@@ -0,0 +1,132 @@
+use didcomm::{Message, PackEncryptedOptions, UnpackOptions};
+
+use crate::{
+    model::stateful::entity::Messages,
+    web::{error::MediationError, AppState, AppStateRepository},
+};
+/// mediator receives messages of type forward then it unpacks the messages and stores it for pickup
+/// the unpacked message is then repacked for further transmission.
+/// Note: Stored messages are not re_packed and must be before transmission in case of
+/// Rewrapping.
+pub async fn mediator_forward_process(
+    payload: &str,
+    state: &AppState,
+) -> Result<Vec<String>, MediationError> {
+    // unpack encrypted payload message
+    let AppStateRepository {
+        message_repository, ..
+    } = state
+        .repository
+        .as_ref()
+        .expect("Missing Persistence Layer");
+
+    let mut forward = Vec::new();
+    let mediator_did = &state.diddoc.id;
+    let (unpack_msg, _) = Message::unpack(
+        payload,
+        &state.did_resolver,
+        &state.secrets_resolver,
+        &UnpackOptions::default(),
+    )
+    .await
+    .map_err(|_| MediationError::MessageUnpackingFailure)?;
+    if unpack_msg.to.is_some() {
+        let dids = unpack_msg.clone().to.expect("to field is None");
```

Comment:
No its not :sweat_smile:

Reply:
This approach is not optimal. Instead, you could use the matching `if let Some(data)` pattern. 
Additionally, this check is unnecessary because if the mediator were not the intended recipient, it would not have been able to decrypt the message. What you need to inspect here is the `next` attribute within the `body` field to get the did (or the key) of the recipient.

Reply:
:genie_man: